docs/tools/terraform/reference/cookbook/kafka-clickhouse-integration-recipe.rst

```diff
@@ -0,0 +1,163 @@
+
+  1. ``provider.tf`` file
+
+    .. code-block:: terraform
+
+       terraform {
+         required_providers {
+           aiven = {
+             source  = "aiven/aiven"
+             version = "~> 3.8.1"
+           }
+         }
+       }
+   
+       provider "aiven" {
+         api_token = var.aiven_api_token
+       }
+
+  .. tip::
+
+    You can set environment variable ``AIVEN_TOKEN`` for the ``api_token`` property so that you don't need to pass the ``-var-file`` flag when executing Terraform commands.
+
+  2. ``variables.tf`` file
+
+  Use it for defining the variables to avoid including sensitive information in source control. The ``variables.tf`` file defines the API token, the project name, and the prefix for the service name.
+
+    .. code:: terraform
+
+       variable "aiven_api_token" {
+         description = "Aiven console API token"
+         type        = string
+       }
+   
+       variable "project_name" {
+         description = "Aiven console project name"
+         type        = string
+       }
+
+  3. ``*.tfvars`` file
+
+  Use it to indicate the actual values of the variables so that they can be passed (with the ``-var-file=`` flag) to Terraform during runtime and excluded later on. Configure the ``var-values.tfvars`` file as follows:
+
+    .. code-block:: terraform
+
+       aiven_api_token     = "<YOUR-AIVEN-AUTHENTICATION-TOKEN-GOES-HERE>"
+       project_name        = "<YOUR-AIVEN-CONSOLE-PROJECT-NAME-GOES-HERE>"
+
+Configure the ``services.tf`` file
+''''''''''''''''''''''''''''''''''
+
+The following Terraform script initializes both Aiven for Apache Kafka and Aiven for ClickHouse services, creates the service integration, the source Apache Kafka topic, and the Aiven for ClickHouse database.
+
+.. code-block:: terraform
+
+  resource "aiven_kafka" "kafka" {
+    project                 = var.project_name
+    cloud_name              = "google-europe-west1"
+    plan                    = "business-4"
+    service_name            = "kafka-gcp-eu"
+    maintenance_window_dow  = "monday"
+    maintenance_window_time = "10:00:00"
+  }
+
+  resource "aiven_kafka_topic" "source" {
+    project      = var.project_name
+    service_name = aiven_kafka.kafka.service_name
+    partitions   = 2
+    replication  = 3
+    topic_name   = "iot_measurements"
+  }
+
+  resource "aiven_clickhouse" "clickhouse" {
+    project                 = var.project_name
+    cloud_name              = "google-europe-west1"
+    plan                    = "startup-beta-8"
+    service_name            = "clickhouse-gcp-eu"
+    maintenance_window_dow  = "monday"
+    maintenance_window_time = "10:00:00"
+  }
+
+  resource "aiven_service_integration" "clickhouse_kafka_source" {
+    project                  = var.project_name
+    integration_type         = "clickhouse_kafka"
+    source_service_name      = aiven_kafka.kafka.service_name
+    destination_service_name = aiven_clickhouse.clickhouse.service_name
```

Comment:
This is not complete, to enable the integration user need to specify the target table schema like so
```suggestion
    destination_service_name = aiven_clickhouse.clickhouse.service_name
    clickhouse_kafka_user_config {
      tables {
	name        = "edge_measurements_raw"
	group_name  = "clickhouse-ingestion"
	data_format = "JSONEachRow"
	columns {
	  name = "sensor_id"
	  type = "UInt64"
	}
	columns {
	  name = "ts"
	  type = "DateTime64(6)"
	}
	columns {
	  name = "key"
	  type = "LowCardinality(String)"
	}
	columns {
	  name = "value"
	  type = "Float64"
	}
	topics {
	  name = aiven_kafka_topic.edge_measurements.topic_name
	}
      }
    }

```